Null pointer dereference in X11 initialization error message

### Tested versions

master (9d1cbab1c432b6f1d66ec939445bec68b6af519e)

### System information

n/a

### Issue description

https://github.com/godotengine/godot/blob/9d1cbab1c432b6f1d66ec939445bec68b6af519e/platform/linuxbsd/x11/display_server_x11.cpp#L6032-L6034

### Steps to reproduce

Didn't actually try to trigger this, but my compiler warned about it.

### Minimal reproduction project (MRP)

n/a

I noticed this too, as commented on the PR which introduced it:

@RandomShaper I'm running into this warning with GCC 12.3.0 on Linux when building with `scons warnings=extra werror=yes tests=yes verbose=yes`:
```
In file included from ./core/os/mutex.h:34, 
                 from ./core/os/thread_safe.h:34, 
                 from ./core/object/message_queue.h:35, 
                 from ./core/object/object.h:35, 
                 from ./core/variant/binder_common.h:35, 
                 from ./core/object/method_bind.h:34, 
                 from ./core/object/class_db.h:34, 
                 from ./core/object/ref_counted.h:34, 
                 from ./core/io/resource_uid.h:34, 
                 from ./core/io/resource.h:34, 
                 from ./core/input/input_event.h:35, 
                 from ./core/input/input.h:34, 
                 from platform/linuxbsd/joypad_linux.h:36, 
                 from platform/linuxbsd/x11/display_server_x11.h:36, 
                 from platform/linuxbsd/x11/display_server_x11.cpp:31: 
platform/linuxbsd/x11/display_server_x11.cpp: In constructor 'DisplayServerX11::DisplayServerX11(const String&, DisplayServer::WindowMode, DisplayServer::VSyncMode, uint32_t, const Vector2i*, const Vector2i&, in
t, Error&)': 
platform/linuxbsd/x11/display_server_x11.cpp:6034:45: error: 'this' pointer is null [-Werror=nonnull] 
 6034 |                         ERR_FAIL_MSG(vformat("Could not initialize %s", context_rd->get_api_name())); 
      |                                      ~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ 
./core/error/error_macros.h:595:95: note: in definition of macro 'ERR_FAIL_MSG' 
  595 |                 _err_print_error(FUNCTION_STR, __FILE__, __LINE__, "Method/function failed.", m_msg); \ 
      |                                                                                               ^~~~~

```

Weirdly enough, I don't reproduce it on my main laptop with the same distro and GCC version, with `dev_build=yes dev_mode=yes scu_build=all`.

Haven't looked into what the warning means yet.

--- a/platform/android/display_server_android.cpp
+++ b/platform/android/display_server_android.cpp
@@ -538,9 +538,10 @@ void DisplayServerAndroid::reset_window() {
 #endif
 
 		if (context_rd->window_create(MAIN_WINDOW_ID, last_vsync_mode, display_size.width, display_size.height, &wpd) != OK) {
+			ERR_PRINT(vformat("Failed to reset %s window.", context_rd->get_api_name()));
 			memdelete(context_rd);
 			context_rd = nullptr;
-			ERR_FAIL_MSG(vformat("Failed to reset %s window.", context_rd->get_api_name()));
+			return;
 		}
 	}
 #endif
@@ -575,9 +576,10 @@ DisplayServerAndroid::DisplayServerAndroid(const String &p_rendering_driver, Dis
 
 	if (context_rd) {
 		if (context_rd->initialize() != OK) {
+			ERR_PRINT(vformat("Failed to initialize %s context", context_rd->get_api_name()));
 			memdelete(context_rd);
 			context_rd = nullptr;
-			ERR_FAIL_MSG(vformat("Failed to initialize %s context", context_rd->get_api_name()));
+			return;
 		}
 
 		Size2i display_size = OS_Android::get_singleton()->get_display_size();
@@ -596,9 +598,10 @@ DisplayServerAndroid::DisplayServerAndroid(const String &p_rendering_driver, Dis
 #endif
 
 		if (context_rd->window_create(MAIN_WINDOW_ID, p_vsync_mode, display_size.width, display_size.height, &wpd) != OK) {
+			ERR_PRINT(vformat("Failed to create %s window.", context_rd->get_api_name()));
 			memdelete(context_rd);
 			context_rd = nullptr;
-			ERR_FAIL_MSG(vformat("Failed to create %s window.", context_rd->get_api_name()));
+			return;
 		}
 
 		rendering_device = memnew(RenderingDevice);

--- a/platform/ios/display_server_ios.mm
+++ b/platform/ios/display_server_ios.mm
@@ -87,17 +87,19 @@ DisplayServerIOS::DisplayServerIOS(const String &p_rendering_driver, WindowMode 
 
 	if (context_rd) {
 		if (context_rd->initialize() != OK) {
+			ERR_PRINT(vformat("Failed to initialize %s context", context_rd->get_api_name()));
 			memdelete(context_rd);
 			context_rd = nullptr;
-			ERR_FAIL_MSG(vformat("Failed to initialize %s context", context_rd->get_api_name()));
+			return;
 		}
 
 		Size2i size = Size2i(layer.bounds.size.width, layer.bounds.size.height) * screen_get_max_scale();
 		if (context_rd->window_create(MAIN_WINDOW_ID, p_vsync_mode, size.width, size.height, &wpd) != OK) {
+			ERR_PRINT(vformat("Failed to create %s window.", context_rd->get_api_name()));
 			memdelete(context_rd);
 			context_rd = nullptr;
 			r_error = ERR_UNAVAILABLE;
-			ERR_FAIL_MSG(vformat("Failed to create %s window.", context_rd->get_api_name()));
+			return;
 		}
 
 		rendering_device = memnew(RenderingDevice);

--- a/platform/linuxbsd/x11/display_server_x11.cpp
+++ b/platform/linuxbsd/x11/display_server_x11.cpp
@@ -6028,10 +6028,11 @@ DisplayServerX11::DisplayServerX11(const String &p_rendering_driver, WindowMode 
 
 	if (context_rd) {
 		if (context_rd->initialize() != OK) {
+			ERR_PRINT(vformat("Could not initialize %s", context_rd->get_api_name()));
 			memdelete(context_rd);
 			context_rd = nullptr;
 			r_error = ERR_CANT_CREATE;
-			ERR_FAIL_MSG(vformat("Could not initialize %s", context_rd->get_api_name()));
+			return;
 		}
 		driver_found = true;
 	}

--- a/platform/windows/display_server_windows.cpp
+++ b/platform/windows/display_server_windows.cpp
@@ -4344,10 +4344,11 @@ DisplayServer::WindowID DisplayServerWindows::_create_window(WindowMode p_mode, 
 			}
 #endif
 			if (context_rd->window_create(id, p_vsync_mode, WindowRect.right - WindowRect.left, WindowRect.bottom - WindowRect.top, &wpd) != OK) {
+				ERR_PRINT(vformat("Failed to create %s Window.", context_rd->get_api_name()));
 				memdelete(context_rd);
 				context_rd = nullptr;
 				windows.erase(id);
-				ERR_FAIL_V_MSG(INVALID_WINDOW_ID, vformat("Failed to create %s Window.", context_rd->get_api_name()));
+				return INVALID_WINDOW_ID;
 			}
 			wd.context_created = true;
 		}

